services/catalog/service/src/common/db/in-memory-locales.ts

```diff
@@ -6,17 +6,34 @@ import {
 } from '@axinom/mosaic-service-common';
 import { Client } from 'pg';
 import { all, deletes, insert, Queryable, select } from 'zapatos/db';
+import { locales } from 'zapatos/schema';
 import { Config } from '../config';
-import { MOSAIC_LOCALE_NOTIFY } from '../constants';
+import { DEFAULT_LOCALE_TAG, MOSAIC_LOCALE_NOTIFY } from '../constants';
 
-let inMemoryLocales: string[] = [];
+let inMemoryLocales: locales.JSONSelectable[] = [];
 
 /**
- * Returns a list of locales that currently exist in the system.
- * If an API request is made for a specific locale and it is not in this array -
- * default locale would be used instead.
+ * Returns a language tag for a requested locale tag from the in-memory array of
+ * locales currently registered within the service.
+ * If not found - locale that is marked as default will be returned.
+ * If such locale is not found - fallback to hardcoded `DEFAULT_LOCALE_TAG` is
+ * used.
  */
-export const getInMemoryLocales = (): string[] => inMemoryLocales;
+export const getInMemoryLocale = (locale: string): string => {
+  const found = inMemoryLocales.find(
+    (l) => l.locale.toLowerCase() === locale.toLowerCase(),
+  );
+  if (found) {
+    return found.locale;
+  }
+
+  const def = inMemoryLocales.find((l) => l.is_default);
```

Comment:
minor cosmetics - shall we rename this variable to something like `defaultLocale` or similar  just so it's clear that this is the default locale at the first glance?

Reply:
I was considering it, but I don't like that below it will look like `defaultLocale.locale` :)
And `default` is not a valid variable name... I think `is_default` in find is a good-enough indication

Reply:
I was thinking about `defaultLocale.locale` as well. First thought maybe `defaultSetting` would be a better fit. But then I saw we're actually having some Postgres settings set for locales as well so thought against it.

It's just when I saw `def`, the first that came to my mind was `definition` and took a second to realize it means default. xD

I'll leave it to you to decide if we really want to rename or not. Think we can live with `is_default`

Reply:
Decided to go with `foundDefault`